src/cluster/kv/etcd/store.go

```diff
@@ -616,7 +617,35 @@ func (c *client) writeCacheToFile() error {
 	return nil
 }
 
+func (c *client) createCacheDir() error {
+	path := path.Dir(c.opts.CacheFileFn()(c.opts.Prefix()))
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			if err := os.MkdirAll(path, 0755); err != nil {
```

Comment:
Sorry I was a little vague about this. I think if you use `os.MkdirAll` you can avoid the `os.Stat` and `os.IsNotExist`. If the path is already a directory then it will just return nil so I think the existing checks are redundant.

The permissions thing is gonna be kind of annoying :/

We actually flow the new directory permissions around the codebase as options, so if you take a look at `server.go` you'll see that when we call `lockfile.CreateAndAcquire(lockPath, newDirectoryMode)` (again, you dont need to use that function, just using it as an example that does something similar) the `newDirectoryMode` is passed in based on whatever its configured in our `FilesystemOptions` so I think we'll need to allow setting a `NewDirectoryMode` on whatever options are used to construct this store and then flow that into here if that makes sense.

Reply:
Ahh I didn't get it the first time, sorry about that. Fixed it now.

And the permission thing was a bit annoying indeed haha. I think I got it now but it was quite the treasure hunt and I had to touch a couple of files along my way (tried to recap it in my commit message of 0d53e7e). Let me know if there is anything wrong or if I forgot anything